src/engine/ruleset/decoders/fortinet/fortimail.yml

```diff
@@ -0,0 +1,205 @@
+        type=<event.action> subtype=<~log.rsa.internal.messageid> pri=<log.level> user=<user.name> ui=<~log.rsa.network.network_service> module=<~log.fill> submodule=<~log.fill> msg="<~log.rsa.internal.event_desc>"
+    # date=2016-6-5 time=21:33:08 device_id=ntutl log_id=caecatc log_part=onsequat type=event subtype=update pri=low msg="edquiano"
+    - event.original: >-
+        date=<~log.date> time=<~log.time> device_id=<observer.serial_number> log_id=<event.code> log_part=<~log.rsa.misc.reference_id1>
+        type=<event.action> subtype=<~log.rsa.internal.messageid> pri=<log.level> msg="<~log.rsa.internal.event_desc>"
+
+normalize:
+  - map:
+    - event.dataset: fortinet.fortimail
+    - event.module": fortinet
+    - fileset.name": fortimail
+    - input.type": log
+    - observer.product: FortiMail
+    - observer.type: Firewall
+    - observer.vendor: Fortinet
+    - rsa.email.email_dst: $~log.rsa.email.email_dst
+    - rsa.email.email_src: $~log.rsa.email.email_src
+    - rsa.email.subject: $~log.rsa.email.subject
+    - rsa.email.subject: $~log.rsa.misc.event_type
+    - rsa.internal.event_desc: $~log.rsa.internal.event_desc
+    - rsa.internal.messageid: $event.action
+    - rsa.internal.messageid: $~log.rsa.internal.messageid
+    - rsa.misc.checksum: $~log.rsa.misc.checksum
+    - rsa.misc.client: $~log.rsa.misc.client
+    - rsa.misc.context: $~log.rsa.misc.context
+    - rsa.misc.disposition: $~log.rsa.misc.disposition
+    - rsa.misc.event_type: $event.action
+    - rsa.misc.filter: $~log.rsa.misc.filter
+    - rsa.misc.filter: $~log.rsa.misc.filter
+    - rsa.misc.log_session_id: $~log.rsa.misc.filter
+    - rsa.misc.msgIdPart1: $event.action
+    - rsa.misc.msgIdPart2: $~log.rsa.internal.messageid
+    - rsa.misc.reference_id1: $~log.rsa.misc.reference_id1
+    - rsa.misc.reference_id: $event.code
+    - rsa.misc.result: $~log.rsa.misc.result
+    - rsa.misc.severity: $log.level
+    - rsa.misc.virusname: $~log.rsa.misc.virusname
+    - rsa.network.domain: $server.domain
+    - rsa.web.fqdn: $~log.rsa.web.fqdn
+
+  - check:
+      - rsa.internal.event_desc: +ef_exists
+    logpar:
+      - rsa.internal.event_desc: >-
+          <~log.fill> to \<\<<~log.rsa.email.email_dst>>; reason:<~log.rsa.misc.result>; <~log.fill>
+      - rsa.internal.event_desc: >-
+          to=\<\<<~log.rsa.email.email_dst>>, delay=<~log.fill>, xdelay=<~log.fill>, mailer=<network.protocol>,
+          pri=<~log.fill>, relay=<host.hostname>[<source.ip>], <~log.fill>
+      - rsa.internal.event_desc: >-
+          to=\<\<<~log.rsa.email.email_dst>, delay=<~log.fill>, xdelay=<~log.fill>, mailer=<network.protocol>,
+          pri=<~log.fill>, relay=<host.hostname>
+    map:
+      - rsa.misc.result: $~log.rsa.misc.result
+      - rsa.email.email_dst: $~log.rsa.email.email_dst
+      - rsa.internal.event_desc: +ef_delete
+
+  - check:
+      - rsa.web.fqdn: +ef_exists
```

Comment:
This check is redundant, every helper that uses a reference implicitly ensures it exists. This makes decoders longer and we check the field two times.